node/rpc/src/chain_api.rs

```diff
@@ -102,21 +100,17 @@ where
             )
             .await
     } else {
-        let file = File::from_std(tmp_file.reopen()?);
+        let file = tokio::fs::File::create(&temp_path).await?;
         data.chain_store
             .export(
                 &start_ts,
                 recent_roots,
-                AsyncWriterWithChecksum::<Sha256, _>::new(BufWriter::new(file)),
+                AsyncWriterWithChecksum::<Sha256, _>::new(BufWriter::new(file.compat())),
             )
             .await
     } {
         Ok(checksum) if !dry_run => {
-            if let Err(e) = tmp_file.persist(&output_path) {
-                // Temporary files cannot be persisted across filesystems, so we fallback to a
-                // copy in case it happens.
-                tokio::fs::copy(e.file, &output_path).await?;
-            }
+            temp_path.persist(&output_path)?;
```

Comment:
Maybe we should write a note here saying that this rename is expected to succeed because we make sure the temp-file is in the same folder as the final file.

Reply:
Fixed